src/main/scala/org/openeo/opensearch/OpenSearchResponses.scala

```diff
@@ -265,25 +275,33 @@ object OpenSearchResponses {
 
             if(id.endsWith(".SAFE")){
               val all_links = getFilePathsFromManifest(id)
-              Feature(id, extent, nominalDate, all_links.toArray, resolution,tileID,Option(theGeometry))
+              Feature(id, extent, nominalDate, all_links.toArray, resolution,tileID,Option(theGeometry), None, publishedDate)
             }else if(id.contains("COP-DEM_GLO-30-DGED")){
               val all_links = getDEMPathFromInspire(id)
-              Feature(id, extent, nominalDate, all_links.toArray, resolution,tileID,Option(theGeometry))
+              Feature(id, extent, nominalDate, all_links.toArray, resolution,tileID,Option(theGeometry), None, publishedDate)
             }else{
-              Feature(id, extent, nominalDate, links, resolution,tileID,Option(theGeometry))
+              Feature(id, extent, nominalDate, links, resolution,tileID,Option(theGeometry), None, publishedDate)
             }
-
           }
         }
       }
 
+
+
       implicit val decodeFeatureCollection: Decoder[FeatureCollection] = new Decoder[FeatureCollection] {
         override def apply(c: HCursor): Decoder.Result[FeatureCollection] = {
           for {
             itemsPerPage <- c.downField("properties").downField("itemsPerPage").as[Int]
             features <- c.downField("features").as[Array[Feature]]
           } yield {
-            FeatureCollection(itemsPerPage, features)
+            val featuresDeduped = features
+              .groupBy(x => (x.nominalDate, x.geometry)) // Simple check to assume equality
+              .map(tuple => tuple._2.maxBy(_.publishedDate))
+              .toArray
+            if (featuresDeduped.length < features.length) {
+              logger.debug(s"A duplicate was removed from the results. Most recent published was kept")
```

Comment:
Would it be possible to find and log the feature id's that got removed? This would make the logging more effective. We can also use info logging here instead of debug, as it is quite valuable info.

Reply:
The ID is processed at this point in the code. For example: `/eodata/Sentinel-1/SAR/GRD/2021/04/11/S1B_IW_GRDH_1SDV_20210411T054146_20210411T054211_026415_032740_6184.SAFE`

Would printing the published dates also be good?

Like this:
`[main] INFO org.openeo.opensearch.OpenSearchClient - Removing duplicated Features (same startDate and geometry) with the following published dates: Some(2021-04-11T07:55:01.780481Z). Keeping the one with published date: Some(2021-04-11T08:51:07.054814Z)`

Reply:
Or perhaps map the original array to feature id's and also the deduplicated array and then taking the difference?

Reply:
fixed